homeassistant/components/unifi/config_flow.py

```diff
@@ -86,19 +81,26 @@ async def async_step_user(self, user_input=None):
 
         if user_input is not None:
 
-            try:
-                self.config = {
-                    CONF_HOST: user_input[CONF_HOST],
-                    CONF_USERNAME: user_input[CONF_USERNAME],
-                    CONF_PASSWORD: user_input[CONF_PASSWORD],
-                    CONF_PORT: user_input.get(CONF_PORT),
-                    CONF_VERIFY_SSL: user_input.get(CONF_VERIFY_SSL),
-                    CONF_SITE_ID: DEFAULT_SITE_ID,
-                }
+            self.config = {
+                CONF_HOST: user_input[CONF_HOST],
+                CONF_USERNAME: user_input[CONF_USERNAME],
+                CONF_PASSWORD: user_input[CONF_PASSWORD],
+                CONF_PORT: user_input.get(CONF_PORT),
+                CONF_VERIFY_SSL: user_input.get(CONF_VERIFY_SSL),
+                CONF_SITE_ID: DEFAULT_SITE_ID,
+            }
 
+            try:
                 controller = await get_controller(self.hass, **self.config)
-
                 sites = await controller.sites()
+
+            except AuthenticationRequired:
+                errors["base"] = "faulty_credentials"
+
+            except CannotConnect:
+                errors["base"] = "service_unavailable"
+
+            else:
```

Comment:
Do we not need the unknown error handling anymore? We usually include that in config flows.

Reply:
Do we? I don't think I've never gotten an issue reported on it. If you see that it should be there for sake of consistency or other reasoning I can of course put it back. I've been looking at it for some time whether to remove it or not and this was its time I suppose :)

Reply:
Ok, if it doesn't happen, we can remove it.